### ISSUE TITLE
Use javax.security.PrivateKey and PublicKey as argument to the PrivateKey and PublicKey class wrappers

### ISSUE THREAD

**Issue by nbaars** [nbaars/paseto4j#38]
At the moment the moment both classes take a `byte[]` as argument:

```
 public PublicKey(byte[] keyMaterial, Version version) {
        super(keyMaterial, version);
    }
```

It is easier from a user perspective to change this to:

```
 public PublicKey(javax.security.PublicKey, Version version) {
        super(keyMaterial, version);
    }
```

this way the API is not responsible for translations, we take a valid key which the user needs to obtain from a `byte[]`, PEM encoded string etc.

**Comment by JonathanXu8** [comment]
Hi, is this issue still open?

**Comment by nbaars** [comment]
@JonathanXu8 yes, this issue is still open. Let me know if you like to pick it up.

**Comment by JonathanXu8** [comment]
Sounds good! I'd like to pick it up.

**Comment by JonathanXu8** [comment]
Hi @nbaars, just like to confirm the requirements. It seems that V3 already supports public PublicKey(javax.security.PublicKey, Version version). So all I have to do is to implement the same thing for v1 and v2. Right?

**Comment by nbaars** [comment]
Hi @JonathanXu8 indeed only necessary for V1 and V2

**Comment by nbaars** [comment]
This has been merged, waiting for release
